yarn-project/circuits.js/src/structs/verification_key.ts

```diff
@@ -135,6 +135,68 @@ export class VerificationKeyAsFields {
   }
 }
 
+/**
+ * Provides a 'fields' representation of the AVM's verification key
+ */
+export class AvmVerificationKeyAsFields {
+  constructor(public key: Tuple<Fr, typeof AVM_VERIFICATION_KEY_LENGTH_IN_FIELDS>, public hash: Fr) {}
+
+  public get numPublicInputs() {
+    return Number(this.key[CIRCUIT_PUBLIC_INPUTS_INDEX]);
+  }
+
+  public get circuitSize() {
+    return Number(this.key[CIRCUIT_SIZE_INDEX]);
+  }
+
+  public get isRecursive() {
+    return this.key[CIRCUIT_RECURSIVE_INDEX] == Fr.ONE;
```

Comment:
This makes me a bit nervous. Can we use `.equals` instead, since it's overloaded? And yeah, I'm aware this was copied from something that's already wrong, damn. Can you fix the original one as well?